### PR TITLE
Kat Perreira Edges

### PR DESCRIPTION
**Pull request by kat-perreira**
I only did one implementation of this one: I think the time complexity is O(n), linear because
as the number increases in size so does the time at a constant rate. I think the space complexity is constant  O(1) because it takes the same amount of space no matter how big the number gets.

### CODE REVIEW

File: lib/factorial.rb [hunk 1]
```diff
@@ -1,4 +1,22 @@
...
+  end
+
+  # factorial: ! means factorial. n! = n * (n-1) if n > 0
+  # example: 5! = 5 * 4 * 3 * 2 * 1
+  n = number
+  while n > 1
+    n *= number - 1
+    n -= 1
+  end
+  return answer
```
Comment (shrutivanw) [inline lib/factorial.rb:20]: answer seems to be an undefined variable.

File: lib/factorial.rb [hunk 2]
```diff
@@ -1,4 +1,22 @@
...
+
+  # edge case test, number is 0
+  if number == 0
+    return 1
+  end
+
+  # factorial: ! means factorial. n! = n * (n-1) if n > 0
+  # example: 5! = 5 * 4 * 3 * 2 * 1
+  n = number
+  while n > 1
```
Comment (shrutivanw) [inline lib/factorial.rb:16]: This could result in an infinite loop. Are the mini-tests provided passing for you?

**Comment by shrutivanw** [comment]
Based on your explanation of time and space complexity and reviewing the code, I can see what algorithm you're looking to implement. However the code has some bugs. I recommend trying out your algorithm using paper and pen and then coding it. Run the tests to confirm if you got the algorithm right.
